Clinical-Genomics/scout
# Add option for admins to create beacon datasets from institute settings

It is not possible to submit a case to the beacon is the institute doesn't already have a dataset in beacon. Dataset IDs are something like this `custXXX_GRCh37` or `custXXX_GRCh38`, so it's a controlled dictionary.

Fix #3299 and fix #3276. Scout admins will be able to create a new dataset for ai institute (with controlled dictionary) from the institute settings page:

![image](https://user-images.githubusercontent.com/28093618/164428034-1bd4d5d7-a4b0-459d-a81a-d25c0c2bc4e6.png)

The option will not be visible for non-admin users

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
</details>

**Before testing**
Your user should have beacon_submitter role in scout-stage database. If it doesn't:
1. ssh to hasta
2. "us"
3. `scout update user -u <your.email@scilifelab.se> -r beacon_submitter`
4. If you are not an admin: `scout update user -u <your.email@scilifelab.se> -r admin`

**How to test**:
1. Install on scout-stage (cg-vm1)
5. Go to a case from cust000 (doesn't have any dataset registered in beacon)
6. Try to add to beacon a recent case from this institute, you should get an error message.
7. Go to cust004 institute settings and create a new beacon dataset id for data in genome build 37
8. Try again to add variants of a cust000 case to the beacon and now it should work.

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
